corehq/apps/export/forms.py

```diff
@@ -1,139 +1,375 @@
+                ng_required='false',
+            ),
+            crispy.Div(
                 crispy.Field(
-                    'application',
-                    data_bind='value: appId',
-                ),
-                crispy.Div(
-                    crispy.Field(
-                        'case_type',
-                        data_bind=(
-                            "options: caseTypeOptions, "
-                            "optionsText: 'text', "
-                            "optionsValue: 'value', "
-                            "value: case_type"
-                        ),
-                    ),
-                    data_bind="visible: appId",
+                    'user_types',
+                    ng_model='formData.user_types',
+                    ng_required='false',
                 ),
+                ng_show="formData.type_or_group === 'type'",
             ),
             crispy.Div(
-                FormActions(
-                    crispy.ButtonHolder(
-                        crispy.Submit(
-                            'create_export',
-                            _('Next'),
+                B3MultiField(
+                    _("Group"),
+                    crispy.Div(
+                        crispy.Div(
+                            InlineField(
+                                'group',
+                                ng_model='formData.group',
+                                ng_required='false',
+                                style="width: 98%",
+                            ),
+                            ng_show="hasGroups",
                         ),
                     ),
+                    CrispyTemplate('export/crispy_html/groups_help.html', {
+                        'domain': self.domain_object.name,
+                    }),
                 ),
-                data_bind="visible: case_type",
+                ng_show="formData.type_or_group === 'group'",
             ),
+            *self.extra_fields
+        )
+
+    @property
+    def extra_fields(self):
+        """
+        :return: a list of extra crispy.Field classes as additional filters
+        depending on type of export.
+        """
+        return []
+
+    def _get_filtered_users(self):
+        user_types = self.cleaned_data['user_types']
+        user_filter_toggles = [
+            self._USER_MOBILE in user_types,
+            self._USER_DEMO in user_types,
+            self._USER_ADMIN in user_types,
+            self._USER_UNKNOWN in user_types,
+            self._USER_SUPPLY in user_types
+        ]
+        # todo refactor HQUserType
+        user_filters = HQUserType._get_manual_filterset(
+            (True,) * HQUserType.count,
+            user_filter_toggles
+        )
+        return users_matching_filter(self.domain_object.name, user_filters)
+
+    def _get_group(self):
+        group = self.cleaned_data['group']
+        if group:
+            return Group.get(group)
+
+    def get_edit_url(self, export):
```

Comment:
I recently learned about python's [abstract base classes](https://docs.python.org/2/library/abc.html) which I think could be nice in terms of providing static analysis of missing subclass overrides. would be nice to see that used here (and throughout these views)